src/az-purview/deploy/terraform/main.tf

```diff
@@ -0,0 +1,188 @@
+# Configure the Azure provider
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.41.0"
+    }
+
+  }
+
+  required_version = "1.3.2"
+}
+
+// Current Subscription
+data "azurerm_subscription" "default" {}
+
+// Subscription for existing Private DNS Zones, ignore if "new_or_existing_dns_zones" variable is set to "new"
+data "azurerm_subscription" "ops_subscription" {
+  count           = var.new_or_existing_dns_zones == "new" ? 0 : 1
+  subscription_id = var.dns_zone_resource_group_subscription_id
+}
+
+// Default provider
+provider "azurerm" {
+  features {}
+}
+
+// Provider for the recipe
+provider "azurerm" {
+  features {}
+  alias           = "spoke"
+  subscription_id = data.azurerm_subscription.default.subscription_id
+}
+
+// Provider for the Private DNS Zones' subscription
+// - If "new_or_existing_dns_zones" variable is set to "new", use the current subscription id
+// - Otherwise, use the existing DNS Zone's subscription id (variable "dns_zone_resource_group_subscription_id")
+provider "azurerm" {
+  features {}
+  alias           = "hub"
+  subscription_id = var.new_or_existing_dns_zones == "new" ? data.azurerm_subscription.default.subscription_id : data.azurerm_subscription.ops_subscription[0].subscription_id
+}
+
+// Resouce group to deploy the recipe
+resource "azurerm_resource_group" "spoke" {
```

Comment:
The "hub" and "spoke" approach here is a bit different than the other Terraform recipes. It may be an approach worth considering for an update to the others.  :)

Reply:
Not sure I understand the comment here: my understanding was these recipes are spokes that can peer or not to a given hub. Do you want me to change the terraform resource's name?